redfish-core/lib/oem/ibm/lamp_test.hpp

```diff
@@ -0,0 +1,126 @@
+#pragma once
+
+#include "async_resp.hpp"
+#include "dbus_utility.hpp"
+#include "redfish_util.hpp"
+
+#include <array>
+#include <memory>
+#include <string_view>
+#include <variant>
+
+namespace redfish
+{
+
+/**
+ * @brief Retrieves lamp test state.
+ *
+ * @param[in] asyncResp     Shared pointer for generating response message.
+ *
+ * @return None.
+ */
+inline void
+    getLampTestState(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp)
+{
+    BMCWEB_LOG_DEBUG("Get lamp test state");
+
+    std::array<std::string_view, 1> interfaces = {
+        "xyz.openbmc_project.Led.Group"};
+    dbus::utility::getDbusObject(
+        "/xyz/openbmc_project/led/groups/lamp_test", interfaces,
+        [asyncResp](const boost::system::error_code& ec,
+                    const dbus::utility::MapperGetObject& object) {
+        if (ec || object.empty())
+        {
+            if (ec.value() == 5) // generic:5 error
+            {
+                BMCWEB_LOG_DEBUG("lamp test not available yet!!");
+                return;
+            }
+            BMCWEB_LOG_ERROR("DBUS response error: {}", ec.message());
```

Comment:
I think bmcweb suggests to use `ec.value()` for logging.

Reply:
Done